warehouse/query-core/src/main/java/datawave/query/util/sortedmap/FileSerializableSortedMap.java

```diff
@@ -0,0 +1,287 @@
+        }
+
+        @Override
+        public void deleteFile() {
+            delegate.deleteFile();
+        }
+    }
+
+    public static class SerializableInputStream<K extends Serializable,V extends Serializable> implements SortedMapInputStream<K,V> {
+        private final InputStream stream;
+        private ObjectInputStream delegate;
+        private final long length;
+
+        public SerializableInputStream(InputStream stream, long length) throws IOException {
+            this.stream = stream;
+            this.length = length;
+        }
+
+        private ObjectInputStream getDelegate() throws IOException {
+            if (delegate == null) {
+                this.delegate = new ObjectInputStream(stream);
+            }
+            return delegate;
+        }
+
+        @Override
+        public Map.Entry<K,V> readObject() throws IOException {
+            try {
+                K key = (K) getDelegate().readObject();
+                V value = (V) getDelegate().readObject();
+                return new UnmodifiableMapEntry<>(key, value);
+            } catch (IOException ioe) {
+                return null;
+            } catch (ClassNotFoundException nnfe) {
+                return null;
+            }
+        }
+
+        @Override
+        public int readSize() throws IOException {
+            long bytesToSkip = length - 4;
+            long total = 0;
+            long cur = 0;
+
+            while ((total < bytesToSkip) && ((cur = stream.skip(bytesToSkip - total)) > 0)) {
+                total += cur;
+            }
+
+            byte[] buffer = new byte[4];
+            stream.read(buffer);
+
+            return ((buffer[3] & 0xFF)) + ((buffer[2] & 0xFF) << 8) + ((buffer[1] & 0xFF) << 16) + ((buffer[0]) << 24);
+        }
+
+        @Override
+        public void close() {
+            try {
+                if (delegate != null) {
+                    delegate.close();
+                } else {
+                    stream.close();
+                }
+            } catch (Exception e) {
+                log.error("Failed to close input stream", e);
+            }
+        }
+    }
+
+    public static class SerializableOutputStream<K extends Serializable,V extends Serializable> implements FileSortedMap.SortedMapOutputStream<K,V> {
+        private ObjectOutputStream delegate;
+
+        public SerializableOutputStream(OutputStream stream) throws IOException {
+            delegate = new ObjectOutputStream(stream);
+        }
+
+        @Override
+        public void writeObject(K key, V value) throws IOException {
+            delegate.writeObject(key);
+            delegate.writeObject(value);
+        }
+
+        @Override
+        public void writeSize(int size) throws IOException {
+            delegate.write((size >>> 24) & 0xFF);
+            delegate.write((size >>> 16) & 0xFF);
+            delegate.write((size >>> 8) & 0xFF);
+            delegate.write((size >>> 0) & 0xFF);
+        }
```

Comment:
this could be simplified with the above to just write the integer?

Reply:
At one point I was having issues with little/big endian so this guarentees that I am reading and writing the size correctly.